server/web/session.go

```diff
@@ -67,6 +67,16 @@ func loginCheckMiddleware(in echo.HandlerFunc) echo.HandlerFunc {
 		if isLoggedIn {
 			return in(c)
 		}
-		return c.NoContent(http.StatusForbidden)
+
+		errorMessage := "Access restricted, please log in first"
+		errorResponse := struct {
```

Comment:
No need for that, as a string is a valid JSON. You could return `msg` and pass it to render.

Reply:
Done.

Reply:
Do you mean something like that:

```
return c.JSON(http.StatusForbidden, map[string]string{"error": msg})
```

Reply:
I meant:

```go
return c.JSON(http.StatusForbidden, msg)
```

Reply:
The reason I started this is because [JSONLint](https://jsonlint.com/) says it is valid JSON :confused: 

https://jsonlint.com/?json=%22Access%20restricted,%20please%20log%20in%20first%22

If you're sure it is not valid JSON, let's move on.